src/sinks/file/mod.rs

```diff
@@ -683,15 +689,168 @@ mod tests {
+            compression: Compression::None,
+            acknowledgements: Default::default(),
+            timezone: Default::default(),
+            internal_metrics: FileInternalMetricsConfig {
+                include_file_tag: true,
+            },
+        };
+
+        let metric_count = 3;
+        let timestamp = Utc::now().trunc_subsecs(3);
+        let timestamp_offset = Duration::from_secs(1);
+
+        let (input, _events) = random_metrics_with_stream_timestamp(
+            metric_count,
+            None,
+            None,
+            timestamp,
+            timestamp_offset,
+        );
+
+        run_assert_sink(&config, input.clone().into_iter()).await;
+
+        let output = (0..metric_count).map(|index| {
+            let expected_timestamp = timestamp + (timestamp_offset * index as u32);
+            let expected_filename =
+                directory.join(&format!("{}.log", expected_timestamp.format(format)));
+
+            lines_from_file(expected_filename)
+        });
+        for (input, output) in input.iter().zip(output) {
+            // The format will partition by second and metrics are a second apart.
+            assert_eq!(
+                output.len(),
+                1,
+                "Expected the output file to contain one metric"
+            );
+            let output = &output[0];
+
+            let metric_name = input.as_metric().name();
+            assert!(output.contains(metric_name));
+        }
+    }
+
+    // The TestSerializer only supports Log and Metric data types. This test asserts the current state of running the vector file sink with an event stream.
+    // A file partition is created and empty lines are written in the file. Running a real vector instance would fail on the configuration because
+    // it would perform an input validation against the output type.
+    #[tokio::test]
+    async fn trace_single_partition_not_supported() {
+        let template = temp_file();
+
+        let config = FileSinkConfig {
+            path: template.clone().try_into().unwrap(),
+            idle_timeout: default_idle_timeout(),
+            encoding: (None::<FramingConfig>, TextSerializerConfig::default()).into(),
+            compression: Compression::None,
+            acknowledgements: Default::default(),
+            timezone: Default::default(),
+            internal_metrics: FileInternalMetricsConfig {
+                include_file_tag: true,
+            },
+        };
+        assert!((config.input().data_type() & DataType::Trace).is_none());
+
+        let (input, _events) = random_lines_with_stream(100, 64, None);
+
+        run_assert_trace_sink(&config, input.clone()).await;
+
+        let output = lines_from_file(template);
+        assert_eq!(input.len(), output.len());
+
+        for (input, output) in input.into_iter().zip(output) {
+            assert_ne!(input, output);
+            assert_eq!("", output);
+        }
+    }
```

Comment:
Hi @nionata, I appreciate you adding these tests. Thank you. I think we can just remove this one because it expected to only wire components with matching input and output types. Enhancing the `TextSerializerConfig` is definitely orthogonal to this PR.
```suggestion
```

Reply:
done